### PR TITLE
more edits from reviewing tutorial

### PR DESCRIPTION
**Pull request by patriknw**

### CODE REVIEW

File: docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjectionHandler.java [hunk 1]
```diff
@@ -28,12 +29,14 @@ public ItemPopularityProjectionHandler(String tag, ItemPopularityRepository repo
...
       ShoppingCart.ItemAdded added = (ShoppingCart.ItemAdded) event;
       dbEffect = this.repo.update(added.itemId, added.quantity);
+    // end::handler[]
     } else if (event instanceof ShoppingCart.ItemQuantityAdjusted) {
       ShoppingCart.ItemQuantityAdjusted adjusted = (ShoppingCart.ItemQuantityAdjusted) event;
       dbEffect = this.repo.update(adjusted.itemId, adjusted.newQuantity - adjusted.oldQuantity);
     } else if (event instanceof ShoppingCart.ItemRemoved) {
       ShoppingCart.ItemRemoved removed = (ShoppingCart.ItemRemoved) event;
       dbEffect = this.repo.update(removed.itemId, -removed.oldQuantity);
+    // tag::handler[]
```
Comment (patriknw) [inline docs-source/docs/modules/microservices-tutorial/examples/04-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjectionHandler.java:39]: ItemQuantityAdjusted and ItemRemoved are optional exercises in previous steps so can't be included in later snippets

**Review by chbatey — APPROVED** [review]

File: docs-source/docs/modules/microservices-tutorial/examples/05-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjection.java [hunk 2]
```diff
@@ -47,3 +48,4 @@ public static void init(ActorSystem<?> system, ItemPopularityRepository reposito
         () -> new ItemPopularityProjectionHandler(tag, repository));
   }
 }
+// end::projection[]
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/examples/05-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjection.java:51]: (unrelated to this PR). 

Just noticed this snippet (the java version) includes `5` callouts, while the scala equivalent and the writeup have 6 callouts:

Screenshot 2020-10-23 at 10 02 32

File: docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc [hunk 3]
```diff
@@ -9,13 +9,13 @@ In this part, we will add another Projection from the events of the `ShoppingCar
 [caption=""]
 image::example-projection-kafka.png[Example Kafka]
 
+As described in xref:concepts:internal-and-external-communication.adoc[Internal and External Communication concepts] we can publishing messages to a broker, such as Apache Kafka, to decouple communication between different Microservices.
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:12]:
```suggestion
As described in xref:concepts:internal-and-external-communication.adoc[Internal and External Communication concepts] we can publish messages to a broker, such as Apache Kafka, to decouple communication between different Microservices.
```

File: docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc [hunk 4]
```diff
@@ -87,28 +89,30 @@ The serialization converts the `ShoppingCart.Event` classes to the Protobuf repr
...
 +
 [source,java,indent=0]
 ----
 include::example$05-shopping-cart-service-java/src/main/java/shopping/cart/PublishEventsProjection.java[]
 ----
 
 Scala::
+src/main/scala/shopping/cart/PublishEventsProjection.scala:
 +
 [source,scala,indent=0]
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:108]:
```suggestion
[.tabset]
Java::
+
[source,java,indent=0]
.src/main/java/shopping/cart/PublishEventsProjection.java
----
include::example$05-shopping-cart-service-java/src/main/java/shopping/cart/PublishEventsProjection.java[]
----

Scala::
+
[source,scala,indent=0]
.src/main/scala/shopping/cart/PublishEventsProjection.scala
```

Prefer asciidoc metadata syntax

Comment (ignasi35) [reply docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:108]: I only moved the file name and prefixed it with a `.`(dot). Github's `diff` highlights too much.

File: docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc [hunk 5]
```diff
@@ -169,21 +173,23 @@ Note that different services should not share code, but the Protobuf specificati
 
 === Consumer
 
-Create a `ShoppingCartEventConsumer` object that runs an Akka Stream with a Kafka `Consumer.committableSource` from Alpakka Kafka.
+Create a `ShoppingCartEventConsumer` [.group-scala]#object# [.group-java]#class# in `shopping-analytics-service`. It runs an Akka Stream with a Kafka `Consumer.committableSource` from Alpakka Kafka.
 
 [.tabset]
 Java::
+src/main/java/shopping/analytics/ShoppingCartEventConsumer.java:
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:180]: Same, prefer asiidoc syntax.

File: docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc [hunk 6]
```diff
@@ -291,7 +299,7 @@ ItemAdded: 1 pencil to cart cart1
 
 === Stop the service
 
-When finished, stop the service with `ctrl-c`. Leave Cassandra and Kafka running for the next set of steps, or stop it with:
+When finished, stop the `shopping-cart-service` and `shopping-analytics-service` with `ctrl-c`. Leave Cassandra and Kafka running for the next set of steps, or stop it with:
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-kafka.adoc:302]:
```suggestion
When finished, stop the `shopping-cart-service` and `shopping-analytics-service` with `ctrl-c`. Leave Cassandra and Kafka running for the next set of steps, or stop them with:
```

File: docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc [hunk 7]
```diff
@@ -68,13 +64,15 @@ The `ItemPopularityRepository` encapsulates the database access and can have a s
 
 [.tabset]
 Java::
+src/main/java/shopping/cart/ItemPopularityRepository.java:
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc:67]: asciidoc syntax (dot-prefix in the `[source]` block)

File: docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc [hunk 8]
```diff
@@ -83,13 +81,15 @@ include::example$04-shopping-cart-service-scala/src/main/scala/shopping/cart/Ite
...
 Java::
+src/main/java/shopping/cart/ItemPopularityRepositoryImpl.java:
 +
 [source,java,indent=0]
 ----
 include::example$04-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityRepositoryImpl.java[]
 ----
 
 Scala::
+src/main/scala/shopping/cart/ItemPopularityRepository.scala:
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc:92]: same

File: docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc [hunk 9]
```diff
@@ -151,22 +152,25 @@ include::example$04-shopping-cart-service-scala/src/main/scala/shopping/cart/Sho
...
+src/main/java/shopping/cart/ItemPopularityProjection.java:
 +
 [source,java,indent=0]
 ----
-include::example$04-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjection.java[]
+include::example$04-shopping-cart-service-java/src/main/java/shopping/cart/ItemPopularityProjection.java[tag=projection]
 ----
 
 Scala::
+src/main/scala/shopping/cart/ItemPopularityProjection.scala:
```
Comment (ignasi35) [inline docs-source/docs/modules/microservices-tutorial/pages/projection-query.adoc:169]: same

**Review by ignasi35 — APPROVED** [review]
Quite the cleanup! 
Left a few small comments.

**Comment by patriknw** [comment]
Thanks for comments @ignasi35, I will incorporate in next PR since this is green.

**Pull request merged**